docs/knative-rabbitmq.md

```diff
@@ -0,0 +1,335 @@
+# Setup Knative with RabbitMQ
+
+No need to use kind for cluster
+
+```
+brew install kind
+```
+
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+name: treetracker-knative
+nodes:
+- role: control-plane
+  extraPortMappings:
+    ## expose port 31380 of the node to port 80 on the host
+  - containerPort: 31080
+    hostPort: 80
+    ## expose port 31443 of the node to port 443 on the host
+  - containerPort: 31443
+    hostPort: 443
+```
+
+```sh
+kind create cluster --config kind-cluster.yaml
+```
+
+## RabbitMQ
+
+### RabbitMQ cluster operator
+
+```sh
+kubectl apply -f https://github.com/rabbitmq/cluster-operator/releases/latest/download/cluster-operator.yml
+```
+
+### RabbitMQ cluster
+
+```yaml
+# cluster.yaml
+apiVersion: rabbitmq.com/v1beta1
+kind: RabbitmqCluster
+metadata:
+  name: rabbitmq-cluster
+spec:
+  replicas: 1
+```
+
+```sh
+kubectl apply -f cluster.yaml
+```
+
+Wait until you see the pod is running
+
+```sh
+kubectl get pods -w
+```
+
+## RabbitMQ plugin(Optional)
+
+If you want to streamline the configuration process, you can install a convenient RabbitMQ plugin via krew:
+
+```sh
+kubectl krew install rabbitmq
+```
+
+List cluster
+
+```sh
+kubectl rabbitmq list
+```
+
+To find your RabbitMQ credentials:
+
+```
+kubectl rabbitmq secrets rabbitmq-cluster
+```
+
+Open the port for RabbitMQ's management interface:
+
+```sh
+kubectl rabbitmq manage rabbitmq-cluster
+```
+
+Then, navigate to [https://localhost:15672](https://localhost:15672/) and login with the credentials above.
+
+## Create Queue
+
+
+### Install cert-manager
+
+```yaml
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.3.1/cert-manager.yaml
+```
+
+### Install message topology
+
+```sh
+kubectl apply -f https://github.com/rabbitmq/messaging-topology-operator/releases/latest/download/messaging-topology-operator-with-certmanager.yaml
+```
+
+
+## RabbitMQ controller
+
+```
+kubectl apply -f https://github.com/knative-sandbox/eventing-rabbitmq/releases/download/knative-v1.11.1/rabbitmq-broker.yaml
+```
+
+## Knative Serving
+
+```sh
+kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.11.0/serving-crds.yaml
+```
+
+```sh
+kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.11.0/serving-core.yaml
+```
+
+## Install network (kourier)
```

Comment:
This means we don't use Istio?

Reply:
I think we can still use Istio if prefered but I haven't try out that.

Reply:
I'd like to leave this to you to decide, because you are the one most understand the case now.